spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/datastore/health/DatastoreHealthIndicatorAutoConfigurationTests.java

```diff
@@ -0,0 +1,82 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.datastore.health;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.datastore.Datastore;
+import org.junit.Test;
+
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.DatastoreRepositoriesAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.DatastoreTransactionManagerAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.GcpDatastoreAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.TestRepository;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreOperations;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTransactionManager;
+import org.springframework.context.annotation.Bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for Datastore health check auto-config.
+ *
+ * @author Raghavan N S
+ * @author Srinivasa Meenavalli
+ * @author Mike Eltsufin
+ *
+ * @since 1.2
+ */
+public class DatastoreHealthIndicatorAutoConfigurationTests {
+
+	private ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(GcpDatastoreAutoConfiguration.class,
+					GcpContextAutoConfiguration.class,
+					DatastoreHealthIndicatorAutoConfiguration.class))
+			.withUserConfiguration(TestConfiguration.class)
+			.withPropertyValues("spring.cloud.gcp.datastore.project-id=test-project",
+					"spring.cloud.gcp.datastore.namespace-id=testNamespace",
+					"management.health.datastore.enabled=true");
+
+
+	@Test
+	public void testDatastoreCreated() {
+		this.contextRunner.run((context) -> assertThat(context.getBean(Datastore.class)).isNotNull());
+	}
+
+	@Test
+	public void testDatastoreHealthIndicatorCreated() {
+		this.contextRunner.run((context) -> assertThat(context.getBean(DatastoreHealthIndicator.class)).isNotNull());
+	}
+
+	/**
+	 * Spring Boot config for tests.
+	 */
+	@AutoConfigurationPackage
```

Comment:
I think this is wrong annotation to indicate a simple `@Configuration`.
What made you to use exactly this annotation instead?

Reply:
It works with `@Configuration`, but interestingly we also use it in other places like `GcpSpannerAutoConfigurationTests`, and plain old `@Configuration` doesn't work in that case. :confused: